app/javascript/dashboard/components/widgets/AttachmentsPreview.vue

```diff
@@ -23,73 +23,59 @@
         </span>
       </div>
       <div class="w-[30%] justify-center">
-        <span
-          class="item overflow-hidden text-xs text-ellipsis whitespace-nowrap"
-        >
+        <span class="overflow-hidden text-xs text-ellipsis whitespace-nowrap">
           {{ formatFileSize(attachment.resource) }}
         </span>
       </div>
       <div class="flex items-center justify-center">
         <woot-button
-          v-if="!isTypeAudio(attachment.resource)"
-          class="remove--attachment clear secondary"
+          class="!w-6 !h-6 text-sm rounded-md hover:bg-slate-50 dark:hover:bg-slate-800 clear secondary"
           icon="dismiss"
-          @click="() => onRemoveAttachment(index)"
+          @click="onRemoveAttachment(index)"
         />
       </div>
     </div>
   </div>
 </template>
-<script>
+
+<script setup>
+import { computed } from 'vue';
 import { formatBytes } from 'shared/helpers/FileHelper';
-export default {
-  props: {
-    attachments: {
-      type: Array,
-      default: () => [],
-    },
-    removeAttachment: {
-      type: Function,
-      default: () => {},
-    },
-  },
-  methods: {
-    onRemoveAttachment(index) {
-      this.removeAttachment(index);
-    },
-    formatFileSize(file) {
-      const size = file.byte_size || file.size;
-      return formatBytes(size, 0);
-    },
-    isTypeImage(file) {
-      const type = file.content_type || file.type;
-      return type.includes('image');
-    },
-    isTypeAudio(file) {
-      const type = file.content_type || file.type;
-      return type.includes('audio');
-    },
-    fileName(file) {
-      return file.filename || file.name;
-    },
+
+const props = defineProps({
+  attachments: {
+    type: Array,
+    default: () => [],
   },
+});
+
+const emits = defineEmits(['remove-attachment']);
+
+const filteredAttachments = computed(() => {
+  return props.attachments.filter(attachment => !attachment?.isRecordedAudio);
+});
+
+const onRemoveAttachment = itemIndex => {
+  const updatedAttachments = filteredAttachments.value.filter(
+    (item, index) => itemIndex !== index
+  );
+  const recordedAudios = props.attachments.filter(
+    attachment => attachment?.isRecordedAudio
+  );
+  emits('remove-attachment', [...updatedAttachments, ...recordedAudios]);
```

Comment:
Instead of `filteredAttachments` we can give a meaningful name like `nonAudioAttachments`.

```
const audioAttachments = computed(() =>
  props.attachments.filter(a => a.isRecordedAudio)
);
const nonAudioAttachments = computed(() =>
  props.attachments.filter(a => !a.isRecordedAudio)
);

const onRemoveAttachment = itemIndex => {
  emits(
    'remove-attachment',
    nonAudioAttachments.value
      .filter((_, index) => index !== itemIndex)
      .concat(audioAttachments.value)
  );
};
```

Reply:
@muhsin-k Done